core/src/main/java/com/google/instrumentation/stats/StatsContextFactory.java

```diff
@@ -35,4 +40,69 @@
    * Returns the default {@link StatsContext}.
    */
   public abstract StatsContext getDefault();
+
+  /**
+   * Get current StatsContext from current gRPC Context.
+   *
+   * @return the current {@code StatsContext} from {@code io.grpc.Context}, or {@code null} if
+   *     there's no {@code StatsContext} associated with current {@code io.grpc.Context}.
+   */
+  @Nullable
+  public final StatsContext getCurrentStatsContext() {
+    // TODO(songya): return a no-op StatsContext (or default) when getCurrentStatsContext() returns
+    // null?
+    return ContextUtils.getCurrentStatsContext();
+  }
+
+  /**
+   * Enters the scope of code where the given {@link StatsContext} is in the current Context, and
+   * returns an object that represents that scope. The scope is exited when the returned object is
+   * closed.
+   *
+   * <p>Supports try-with-resource idiom.
+   *
+   * <p>Example of usage:
+   *
+   * <pre>{@code
+   * // initialized by constructor or by calling Stats.getStatsContextFactory()
+   * private final StatsContextFactory statsCtxFactory;
```

Comment:
Updated, though we still need to check if `statsCtxFactory` is null or not, since [Stats.getStatsContextFactory()](https://github.com/google/instrumentation-java/blob/master/core/src/main/java/com/google/instrumentation/stats/Stats.java#L29) is Nullable.